ui/app/components/http-requests-bar-chart.js

```diff
@@ -0,0 +1,150 @@
+ *
+ * @example
+ * ```js
+ * <HttpRequestsBarChart @counters={{counters}} />
+ * ```
+ *
+ * @param counters=null {Array} - A list of objects containing the total number of HTTP Requests for each month. `counters` should be the response from the `/internal/counters/requests` endpoint which looks like:
+ * COUNTERS = [
+ *    {
+ *       "start_time": "2019-05-01T00:00:00Z",
+ *       "total": 50
+ *     }
+ * ]
+ */
+
+const HEIGHT = 240;
+
+export default Component.extend({
+  classNames: ['http-requests-bar-chart-container'],
+  counters: null,
+  margin: { top: 24, right: 16, bottom: 24, left: 16 },
+  padding: 0.04,
+  width: 0,
+  height() {
+    const { margin } = this;
+    return HEIGHT - margin.top - margin.bottom;
+  },
+
+  parsedCounters: computed('counters', function() {
+    // parse the start times so bars and ticks display properly
+    const { counters } = this;
+    return counters.map(counter => {
+      return assign({}, counter, { start_time: d3TimeFormat.isoParse(counter.start_time) });
+    });
+  }),
+
+  yScale: computed('parsedCounters', 'height', function() {
+    const { parsedCounters } = this;
+    const height = this.height();
+    const counterTotals = parsedCounters.map(c => c.total);
+
+    return d3Scale
+      .scaleLinear()
+      .domain([0, Math.max(...counterTotals)])
+      .range([height, 0]);
+  }),
+
+  xScale: computed('parsedCounters', 'width', function() {
+    const { parsedCounters, width, margin, padding } = this;
+
+    return d3Scale
+      .scaleBand()
+      .domain(parsedCounters.map(c => c.start_time))
+      .rangeRound([0, width - margin.left - margin.right], 0.05)
+      .paddingInner(padding)
+      .paddingOuter(padding);
+  }),
+
+  didInsertElement() {
+    this._super(...arguments);
+    const { margin } = this;
+
+    // set the width after the element has been rendered because the chart axes depend on it.
+    // this helps us avoid an arbitrary hardcoded width which causes alignment & resizing problems.
+    run.schedule('afterRender', this, () => {
+      this.set('width', this.element.clientWidth - margin.left - margin.right);
+      this.renderBarChart();
+    });
+  },
+
+  renderBarChart() {
+    const { margin, width, xScale, yScale, parsedCounters } = this;
+    const height = this.height();
+    const barChartSVG = d3.select('.http-requests-bar-chart');
+    const barsContainer = d3.select('#bars-container');
+
+    // render the chart
+    d3.select('.http-requests-bar-chart')
+      .attr('width', width + margin.left + margin.right)
+      .attr('height', height + margin.top + margin.bottom)
+      .attr('viewBox', `0 0 ${width} ${height}`);
+
+    // scale and render the axes
+    const yAxis = d3Axis
+      .axisRight(yScale)
+      .ticks(3, '.0s')
+      .tickSizeOuter(0);
+    const xAxis = d3Axis
+      .axisBottom(xScale)
+      .tickFormat(d3TimeFormat.timeFormat('%b %Y'))
```

Comment:
Is it possible to split the month and year into their own line? It helps when scanning across the labels. Not sure if that's as simple as adding `<br/>` though.

Reply:
unfortunately the d3 api doesn't appear to allow passing in custom html to the ticks. passing `<br />` renders the html as a string because the ticks render as an svg, and svgs can't render html inside them. i looked into manually appending a `foreignObject` element to the ticks after they render per the suggestion [here](https://stackoverflow.com/questions/7458546/html-in-svg-in-html), but didn't have access to each counter's start time in the scope. :womp: